app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/galleryuplift/GalleryUpliftViewModel.kt

```diff
@@ -0,0 +1,659 @@
+        scope = viewModelScope,
+        getCurrentId = { currentItem.id() },
+        onScoreChanged = {
+            intent {
+                postSideEffect(GallerySideEffect.NotifyResourceScoresChanged)
+            }
+        }
+    )
+
+    private val messageFlow: MutableSharedFlow<Message> = MutableSharedFlow()
+    private val currentItem: GalleryPresenter.GalleryItem
+        get() = galleryItems[container.stateFlow.value.currentPos]
+
+    fun initialize(
+        rootAndFav: RootAndFav,
+        resourcesIds: List<ResourceId>,
+    ) {
+        this.rootAndFav = rootAndFav
+        this.resourcesIds = resourcesIds
+        onFirstViewAttach()
+    }
+
+    fun onPreviewsItemClick() {
+        intent {
+            postSideEffect(GallerySideEffect.ControlVisible(isVisible = true))
+        }
+    }
+
+    fun bindPlainTextView(view: PreviewPlainTextViewHolderUplift) {
+        viewModelScope.launch {
+            view.reset()
+            val item = galleryItems[view.pos]
+            val path = index.getPath(item.id())!!
+            val content = readText(path)
+            content.onSuccess {
+                view.setContent(it)
+            }
+        }
+    }
+
+    fun bindView(view: PreviewImageViewHolderUplift) = viewModelScope.launch {
+        view.reset()
+        val item = galleryItems[view.pos]
+        val path = index.getPath(item.id())!!
+        val placeholder = ImageUtils.iconForExtension(extension(path))
+        view.setSource(placeholder, item.id(), item.metadata, item.preview)
+    }
+
+    fun getKind(pos: Int): Int =
+        galleryItems[pos].metadata.kind.ordinal
+
+    fun onRemoveFabClick() = viewModelScope.launch(NonCancellable) {
+        analytics.trackResRemove()
+        Timber.d(
+            LogTags.GALLERY_SCREEN,
+            buildString {
+                append("[remove_resource] clicked at position ")
+                append("${container.stateFlow.value.currentPos}")
+            }
+        )
+        deleteResource(currentItem.id())
+        galleryItems.removeAt(container.stateFlow.value.currentPos)
+        if (galleryItems.isEmpty()) {
+            intent {
+                postSideEffect(GallerySideEffect.NavigateBack)
+            }
+            return@launch
+        }
+        onTagsChanged()
+        intent {
+            postSideEffect(
+                GallerySideEffect.DeleteResource(
+                    container.stateFlow.value.currentPos
+                )
+            )
+        }
+    }
+
+    private suspend fun deleteResource(resource: ResourceId) {
+        Timber.d(LogTags.GALLERY_SCREEN, "deleting resource $resource")
+        withContext(Dispatchers.IO) {
+            val path = index.getPath(resource)
+            Files.delete(path)
+        }
+        index.updateAll()
+        intent {
+            postSideEffect(GallerySideEffect.NotifyResourceChange)
+        }
+    }
+
+    private fun onFirstViewAttach() {
+        analytics.trackScreen()
```

Comment:
This is a name from the presenter, which does not mean anything. Maybe replace it with `initStorages`?